# 1. Data Extraction

In this notebook we will extract our data and analyse it. For that purpose, we are importing from our library where we define the
```bcrp_dataframe``` dataframe. This function will allows us to use the API interface of the Central Bank of Reserve of Peru (BCRP) to automatically create a pandas dataframe with the necessary codes.

## 1.1 Libraries

We import the necessary libraries, including our own library in the modules file

In [1]:
# Warnings
import warnings
warnings.filterwarnings("ignore")

# Basic Libraries
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import requests
import seaborn as sns
from scipy import stats
from functools import reduce

# Statsmodels
import statsmodels.api as sm
import pmdarima as pmd
from pmdarima.arima import auto_arima
from statsmodels.tsa.api import VAR
from statsmodels.tsa.vector_ar.var_model import VARResults
from statsmodels.tsa.statespace.sarimax import SARIMAX
from statsmodels.tsa.arima.model import ARIMA
from statsmodels.tsa.stattools import adfuller
from statsmodels.tsa.seasonal import STL

# Machine Learning models
from sklearn.ensemble import RandomForestRegressor, RandomForestClassifier
from sklearn.model_selection import train_test_split, GridSearchCV, TimeSeriesSplit
from sklearn.linear_model import Ridge, Lasso, ElasticNet, ElasticNetCV, LinearRegression
from sklearn.linear_model import LinearRegression
from sklearn.metrics import (
    mean_absolute_error,
    mean_squared_error,
    mean_absolute_percentage_error,
    median_absolute_error,
    r2_score,
    precision_score

)

from xgboost import XGBRegressor



In [2]:
# We import our own functions
import sys
sys.path.append('../../..')  # Move two levels up to the project root
from modules.functions import *

## 1.2 Extraction
We define our inputs and apply them the ```bcrp_dataframe``` function in order to obtain the pandas dataframe with the corresponding series

We define the following inputs:

    series     = the code of the series we are going to extract
    start_date = the starting date, when the BCRP starts using the interest rate as a policy measure
    end_date   = December 2019
    freq       = Monthly frequency

### df_1
We can now create the first dataframe with the ```bcrp_dataframe``` function. This dataframe contains all Consumer Price Index variables in monthly % change. This variables are then seasonally adjusted using the 

In [3]:
series     = ['PN01271PM', 'PN01280PM', 'PN01282PM', 'PN01278PM', 'PN09817PM','PN09816PM', 'PN01276PM', 'PN01313PM', 'PN01314PM',  
             'PN01315PM', 'PN09818PM','PN01286PM']
start_date = '2003-09'
end_date   = '2019-12'
freq       = 'Mensual'

In [4]:
df_1 = bcrp_dataframe( series , start_date , end_date , freq )
df_1.head()

Índice de precios Lima Metropolitana (var% mensual) - IPC  \
Fecha                                                                   
2003-09-01                                           0.558598           
2003-10-01                                           0.049032           
2003-11-01                                           0.167685           
2003-12-01                                           0.563951           
2004-01-01                                           0.537447           

            Índice de precios Lima Metropolitana (var% mensual) - IPC Transables  \
Fecha                                                                              
2003-09-01                                           0.205933                      
2003-10-01                                          -0.035055                      
2003-11-01                                           0.243529                      
2003-12-01                                           0.594507                      
2004-01-01                                           0.265543                      

            Índice de precios Lima Metropolitana (var% mensual) - IPC No Transables  \
Fecha                                                                                 
2003-09-01                                           0.822993                         
2003-10-01                                           0.096153                         
2003-11-01                                           0.120950                         
2003-12-01                                           0.534926                         
2004-01-01                                           0.708509                         

            Índice de precios Lima Metropolitana (var% mensual) - IPC Subyacente  \
Fecha                                                                              
2003-09-01                                           0.005689                      
2003-10-01                                          -0.040038                      
2003-11-01                                           0.125742                      
2003-12-01                                           0.127343                      
2004-01-01                                          -0.055834                      

            Índice de precios Lima Metropolitana (var% mensual) - IPC No Subyacente  \
Fecha                                                                                 
2003-09-01                                           1.416850                         
2003-10-01                                           0.164730                         
2003-11-01                                           0.237966                         
2003-12-01                                           1.196907                         
2004-01-01                                           1.379067                         

            Índice de precios Lima Metropolitana (var% mensual) - IPC Alimentos y Energía  \
Fecha                                                                                       
2003-09-01                                           1.024345                               
2003-10-01                                           0.193699                               
2003-11-01                                           0.256361                               
2003-12-01                                           0.898519                               
2004-01-01                                           1.132403                               

            Índice de precios Lima Metropolitana (var% mensual) - IPC Sin Alimentos y Energía  \
Fecha                                                                                           
2003-09-01                                          -0.017156                                   
2003-10-01                                          -0.070649                                   
2003-11-01                                           0.005988 

In [5]:
def get_trend(df, period=12):
    trend_df = pd.DataFrame()

    for col in df.columns:
        stl_result = STL(df[col], period=period).fit()
        trend_df[col] = stl_result.trend

    return trend_df

df_1 = get_trend(df_1)
df_1.head()

Índice de precios Lima Metropolitana (var% mensual) - IPC  \
Fecha                                                                   
2003-09-01                                           0.518834           
2003-10-01                                           0.492247           
2003-11-01                                           0.465462           
2003-12-01                                           0.438413           
2004-01-01                                           0.411138           

            Índice de precios Lima Metropolitana (var% mensual) - IPC Transables  \
Fecha                                                                              
2003-09-01                                           0.470988                      
2003-10-01                                           0.460730                      
2003-11-01                                           0.449419                      
2003-12-01                                           0.437075                      
2004-01-01                                           0.423839                      

            Índice de precios Lima Metropolitana (var% mensual) - IPC No Transables  \
Fecha                                                                                 
2003-09-01                                           0.552064                         
2003-10-01                                           0.514069                         
2003-11-01                                           0.476486                         
2003-12-01                                           0.439183                         
2004-01-01                                           0.402122                         

            Índice de precios Lima Metropolitana (var% mensual) - IPC Subyacente  \
Fecha                                                                              
2003-09-01                                           0.089570                      
2003-10-01                                           0.091267                      
2003-11-01                                           0.092842                      
2003-12-01                                           0.094296                      
2004-01-01                                           0.095620                      

            Índice de precios Lima Metropolitana (var% mensual) - IPC No Subyacente  \
Fecha                                                                                 
2003-09-01                                           1.146795                         
2003-10-01                                           1.078211                         
2003-11-01                                           1.009369                         
2003-12-01                                           0.940109                         
2004-01-01                                           0.870541                         

            Índice de precios Lima Metropolitana (var% mensual) - IPC Alimentos y Energía  \
Fecha                                                                                       
2003-09-01                                           0.974940                               
2003-10-01                                           0.916599                               
2003-11-01                                           0.858081                               
2003-12-01                                           0.799280                               
2004-01-01                                           0.740242                               

            Índice de precios Lima Metropolitana (var% mensual) - IPC Sin Alimentos y Energía  \
Fecha                                                                                           
2003-09-01                                          -0.019098                                   
2003-10-01                                          -0.008282                                   
2003-11-01                                           0.002340 

### df_2
We create the second dataframe with the ```bcrp_dataframe``` function. This dataframe contains rate variables. We use this variables in levels. It is not necessary to differentiate them.

In [6]:
series     = ['PN00493MM', 'PD04722MM']
start_date = '2003-09'
end_date   = '2019-12'
freq       = 'Mensual'

In [7]:
df_2 = bcrp_dataframe( series , start_date , end_date , freq )
df_2.head()

Tasas de interés del Banco Central de Reserva  - Tasa de Encaje  \
Fecha                                                                         
2003-09-01                                          11.014920                 
2003-10-01                                          10.349944                 
2003-11-01                                          11.690608                 
2003-12-01                                          11.391178                 
2004-01-01                                          10.634030                 

            Tasas de interés del Banco Central de Reserva  - Tasa de Referencia de la Política Monetaria  
Fecha                                                                                                     
2003-09-01                                               2.75                                             
2003-10-01                                               2.75                                             
2003-11-01                                               2.50                                             
2003-12-01                                               2.50                                             
2004-01-01                                               2.50

### df_3
We create the third dataframe with the ```bcrp_dataframe``` function. This dataframe contains monetary variables as well as commodities. We differentiate those variables in order to get their monthly % change

In [8]:
series     = ['PN00495MM', 'PN06481IM', 'PN02125PM', 'PN01661XM','PN01662XM','PN01664XM','PN01660XM']
start_date = '2003-09'
end_date   = '2019-12'
freq       = 'Mensual'

In [9]:
df_3 = bcrp_dataframe( series , start_date , end_date , freq )
df_3 = df_3.pct_change()
df_3 = df_3.dropna()
df_3.head()

Emisión primaria y multiplicador (millones S/) - Circulante Desestacionalizado  \
Fecha                                                                                        
2003-10-01                                           0.010225                                
2003-11-01                                           0.011445                                
2003-12-01                                           0.016607                                
2004-01-01                                           0.019490                                
2004-02-01                                           0.017003                                

            Liquidez internacional del BCRP - RIN - Reservas Internacionales Netas (millones US$)  \
Fecha                                                                                               
2003-10-01                                           0.005542                                       
2003-11-01                                           0.050288                                       
2003-12-01                                          -0.010532                                       
2004-01-01                                           0.036233                                       
2004-02-01                                          -0.005743                                       

            Remuneraciones - Remuneración Mínima Vital - Índice Real (base 1994 = 100)  \
Fecha                                                                                    
2003-10-01                                           0.056953                            
2003-11-01                                          -0.001674                            
2003-12-01                                          -0.005608                            
2004-01-01                                          -0.005346                            
2004-02-01                                          -0.010744                            

            Cotizaciones de productos (promedio del periodo) - Trigo - EEUU (US$ por toneladas)  \
Fecha                                                                                             
2003-10-01                                           0.016588                                     
2003-11-01                                           0.115205                                     
2003-12-01                                           0.037146                                     
2004-01-01                                           0.003546                                     
2004-02-01                                          -0.019994                                     

            Cotizaciones de productos (promedio del periodo) - Maíz - EEUU (US$ por toneladas)  \
Fecha                                                                                            
2003-10-01                                          -0.027808                                    
2003-11-01                                           0.075366                                    
2003-12-01                                           0.048843                                    
2004-01-01                                           0.062558                                    
2004-02-01                                           0.081887                                    

            Cotizaciones de productos (promedio del periodo) - Aceite Soya - EEUU (US$ por toneladas)  \
Fecha                                                                                                   
2003-10-01                                           0.181232                                           
2003-11-01                                           0.017022                                           
2003-12-01                                           0.051254                                           
2004-01-01                                           0.026780                                           
200

In [10]:
df = df_1.join(df_2).join(df_3)
df.dropna(inplace=True)
df.head()

Índice de precios Lima Metropolitana (var% mensual) - IPC  \
Fecha                                                                   
2003-10-01                                           0.492247           
2003-11-01                                           0.465462           
2003-12-01                                           0.438413           
2004-01-01                                           0.411138           
2004-02-01                                           0.383700           

            Índice de precios Lima Metropolitana (var% mensual) - IPC Transables  \
Fecha                                                                              
2003-10-01                                           0.460730                      
2003-11-01                                           0.449419                      
2003-12-01                                           0.437075                      
2004-01-01                                           0.423839                      
2004-02-01                                           0.409852                      

            Índice de precios Lima Metropolitana (var% mensual) - IPC No Transables  \
Fecha                                                                                 
2003-10-01                                           0.514069                         
2003-11-01                                           0.476486                         
2003-12-01                                           0.439183                         
2004-01-01                                           0.402122                         
2004-02-01                                           0.365307                         

            Índice de precios Lima Metropolitana (var% mensual) - IPC Subyacente  \
Fecha                                                                              
2003-10-01                                           0.091267                      
2003-11-01                                           0.092842                      
2003-12-01                                           0.094296                      
2004-01-01                                           0.095620                      
2004-02-01                                           0.096795                      

            Índice de precios Lima Metropolitana (var% mensual) - IPC No Subyacente  \
Fecha                                                                                 
2003-10-01                                           1.078211                         
2003-11-01                                           1.009369                         
2003-12-01                                           0.940109                         
2004-01-01                                           0.870541                         
2004-02-01                                           0.800843                         

            Índice de precios Lima Metropolitana (var% mensual) - IPC Alimentos y Energía  \
Fecha                                                                                       
2003-10-01                                           0.916599                               
2003-11-01                                           0.858081                               
2003-12-01                                           0.799280                               
2004-01-01                                           0.740242                               
2004-02-01                                           0.681065                               

            Índice de precios Lima Metropolitana (var% mensual) - IPC Sin Alimentos y Energía  \
Fecha                                                                                           
2003-10-01                                          -0.008282                                   
2003-11-01                                           0.002340                                   
2003-12-01                                           0.012736 

In [11]:
df.tail()

Índice de precios Lima Metropolitana (var% mensual) - IPC  \
Fecha                                                                   
2019-08-01                                           0.155238           
2019-09-01                                           0.151966           
2019-10-01                                           0.148757           
2019-11-01                                           0.145714           
2019-12-01                                           0.142886           

            Índice de precios Lima Metropolitana (var% mensual) - IPC Transables  \
Fecha                                                                              
2019-08-01                                           0.107169                      
2019-09-01                                           0.103574                      
2019-10-01                                           0.100074                      
2019-11-01                                           0.096671                      
2019-12-01                                           0.093373                      

            Índice de precios Lima Metropolitana (var% mensual) - IPC No Transables  \
Fecha                                                                                 
2019-08-01                                           0.182206                         
2019-09-01                                           0.179086                         
2019-10-01                                           0.176011                         
2019-11-01                                           0.173140                         
2019-12-01                                           0.170545                         

            Índice de precios Lima Metropolitana (var% mensual) - IPC Subyacente  \
Fecha                                                                              
2019-08-01                                           0.172176                      
2019-09-01                                           0.168668                      
2019-10-01                                           0.165254                      
2019-11-01                                           0.161942                      
2019-12-01                                           0.158729                      

            Índice de precios Lima Metropolitana (var% mensual) - IPC No Subyacente  \
Fecha                                                                                 
2019-08-01                                           0.120649                         
2019-09-01                                           0.117677                         
2019-10-01                                           0.114708                         
2019-11-01                                           0.112035                         
2019-12-01                                           0.109809                         

            Índice de precios Lima Metropolitana (var% mensual) - IPC Alimentos y Energía  \
Fecha                                                                                       
2019-08-01                                           0.117414                               
2019-09-01                                           0.111047                               
2019-10-01                                           0.104757                               
2019-11-01                                           0.098706                               
2019-12-01                                           0.092980                               

            Índice de precios Lima Metropolitana (var% mensual) - IPC Sin Alimentos y Energía  \
Fecha                                                                                           
2019-08-01                                           0.187309                                   
2019-09-01                                           0.186615                                   
2019-10-01                                           0.185969 

## 1.3 Data Inspection
We inspect the df. We first verify that all values are non-null. The, we apply the ```describe``` function to see the main variables.

In [12]:
df.isna().sum()

Índice de precios Lima Metropolitana (var% mensual) - IPC                                       0
Índice de precios Lima Metropolitana (var% mensual) - IPC Transables                            0
Índice de precios Lima Metropolitana (var% mensual) - IPC No Transables                         0
Índice de precios Lima Metropolitana (var% mensual) - IPC Subyacente                            0
Índice de precios Lima Metropolitana (var% mensual) - IPC No Subyacente                         0
Índice de precios Lima Metropolitana (var% mensual) - IPC Alimentos y Energía                   0
Índice de precios Lima Metropolitana (var% mensual) - IPC Sin Alimentos y Energía               0
Índice de precios Lima Metropolitana (var% mensual) - IPC Alimentos y Bebidas                   0
Índice de precios Lima Metropolitana (var% mensual) - IPC sin Alimentos y Bebidas               0
Índice de precios Lima Metropolitana (var% mensual) - IPC Subyacente Sin Alimentos y Bebidas    0
Índice de precios Li

In [13]:
df.describe()

Índice de precios Lima Metropolitana (var% mensual) - IPC  \
count                                         195.000000           
mean                                            0.236449           
std                                             0.105236           
min                                             0.056201           
25%                                             0.155856           
50%                                             0.240839           
75%                                             0.292942           
max                                             0.508985           

       Índice de precios Lima Metropolitana (var% mensual) - IPC Transables  \
count                                         195.000000                      
mean                                            0.204956                      
std                                             0.104879                      
min                                            -0.015198                      
25%                                             0.132928                      
50%                                             0.189360                      
75%                                             0.265015                      
max                                             0.473341                      

       Índice de precios Lima Metropolitana (var% mensual) - IPC No Transables  \
count                                         195.000000                         
mean                                            0.256093                         
std                                             0.123443                         
min                                             0.027347                         
25%                                             0.165486                         
50%                                             0.260533                         
75%                                             0.325186                         
max                                             0.610059                         

       Índice de precios Lima Metropolitana (var% mensual) - IPC Subyacente  \
count                                         195.000000                      
mean                                            0.233630                      
std                                             0.091512                      
min                                             0.091163                      
25%                                             0.163598                      
50%                                             0.236332                      
75%                                             0.299551                      
max                                             0.449803                      

       Índice de precios Lima Metropolitana (var% mensual) - IPC No Subyacente  \
count                                         195.000000                         
mean                                            0.235540                         
std                                             0.204664                         
min                                            -0.140622                         
25%                                             0.128311                         
50%                                             0.206382                         
75%                                             0.304751                         
max                                             1.078211                         

       Índice de precios Lima Metropolitana (var% mensual) - IPC Alimentos y Energía  \
count                                         195.000000                               
mean                                            0.292706                               
std                                             0.184651                               
min                                            -0.013365                               
25%                                             0.15

We have 195 observation ranging from ```2003-10-01``` to ```2019-12-01```. The mean of monthly % change of all CPI variables is around 0.2. The mean of the lacing rate and the interest rate is 10.7% and 3.67%, respectively. The three monetary variables have a small monthly % change, around 0.01 and 0.001 for Minimum Wage index. 

## 1.4 Data adjustment
We will rename the columns for easier identification of the variables. We will also create a new dataframe with the lags of the variables. 

In [14]:
# New column names
columns = {
    'Índice de precios Lima Metropolitana (var% mensual) - IPC': 'CPI',
    'Índice de precios Lima Metropolitana (var% mensual) - IPC Transables': 'CPI Tradable',
    'Índice de precios Lima Metropolitana (var% mensual) - IPC No Transables': 'CPI Non-Tradable',
    'Índice de precios Lima Metropolitana (var% mensual) - IPC Subyacente': 'CPI Core',
    'Índice de precios Lima Metropolitana (var% mensual) - IPC No Subyacente': 'CPI Non-Core',
    'Índice de precios Lima Metropolitana (var% mensual) - IPC Alimentos y Energía': 'CPI Food and Energy',
    'Índice de precios Lima Metropolitana (var% mensual) - IPC Sin Alimentos y Energía': 'CPI Excluding Food and Energy',
    'Índice de precios Lima Metropolitana (var% mensual) - IPC Alimentos y Bebidas': 'CPI Food and Beverages',
    'Índice de precios Lima Metropolitana (var% mensual) - IPC sin Alimentos y Bebidas': 'CPI Excluding Food and Beverages',
    'Índice de precios Lima Metropolitana (var% mensual) - IPC Subyacente Sin Alimentos y Bebidas': 'CPI Core Excluding Food and Beverages',
    'Índice de precios Lima Metropolitana (var% mensual) - IPC Importado': 'CPI Imported',
    'Índice de precios Lima Metropolitana (var% mensual) - Índice de Precios al por Mayor': 'Wholesale Price Index',
    'Tasas de interés del Banco Central de Reserva  - Tasa de Encaje': 'Reserve Requirement Rate',
    'Tasas de interés del Banco Central de Reserva  - Tasa de Referencia de la Política Monetaria': 'Monetary Policy Rate',
    'Emisión primaria y multiplicador (millones S/) - Circulante Desestacionalizado': 'Circulating Currency Seasonally Adjusted (mill S/)',
    'Liquidez internacional del BCRP - RIN - Reservas Internacionales Netas (millones US$)': 'Net International Reserves (mill $)',
    'Remuneraciones - Remuneración Mínima Vital - Índice Real (base 1994 = 100)': 'Real Minimum Wage (Index)',
    'Cotizaciones de productos (promedio del periodo) - Trigo - EEUU (US$ por toneladas)': 'Wheat (US$ per ton)',
    'Cotizaciones de productos (promedio del periodo) - Maíz - EEUU (US$ por toneladas)': 'Corn  (US$ per ton)',
    'Cotizaciones de productos (promedio del periodo) - Aceite Soya - EEUU (US$ por toneladas)': 'Soybean oil (US$ per ton)',
    'Cotizaciones de productos (promedio del periodo) - Petróleo - WTI (US$ por barriles)': 'Crude oil (US$ per barrel)'  
}

# We rename the columns so they are easier to analyse
df.rename(columns=columns, inplace=True)

In [15]:
df_lags = df.copy()

for variable in df_lags.columns[1:]:
    df_lags[f'{variable}_lag_1'] = df_lags[variable].shift()
    df_lags[f'{variable}_lag_2'] = df_lags[variable].shift(2)

In [16]:
# We delete contemporary variables
df_lags.drop(columns = ['CPI Tradable', 'CPI Non-Tradable', 'CPI Core', 'CPI Non-Core', 'CPI Food and Energy', 'CPI Excluding Food and Energy',
       'CPI Food and Beverages', 'CPI Excluding Food and Beverages','CPI Core Excluding Food and Beverages', 'CPI Imported',
       'Wholesale Price Index', 'Reserve Requirement Rate','Monetary Policy Rate','Circulating Currency Seasonally Adjusted (mill S/)',
       'Net International Reserves (mill $)', 'Real Minimum Wage (Index)', 'Wheat (US$ per ton)', 'Corn  (US$ per ton)', 
       'Soybean oil (US$ per ton)', 'Crude oil (US$ per barrel)'], inplace = True)

df_lags = df_lags.dropna()

## 1.5 Save Results
We save it to the ```input``` folder, where we can use it to do the forecasting in the next notebook.

In [17]:
df.to_csv('../../../input/df_raw_h19.csv')

In [18]:
df_lags.to_csv('../../../input/df_lags_h19.csv')

In [19]:
df_lags.tail()

CPI  CPI Tradable_lag_1  CPI Tradable_lag_2  \
Fecha                                                          
2019-08-01  0.155238            0.110871            0.114674   
2019-09-01  0.151966            0.107169            0.110871   
2019-10-01  0.148757            0.103574            0.107169   
2019-11-01  0.145714            0.100074            0.103574   
2019-12-01  0.142886            0.096671            0.100074   

            CPI Non-Tradable_lag_1  CPI Non-Tradable_lag_2  CPI Core_lag_1  \
Fecha                                                                        
2019-08-01                0.185235                0.188145        0.175751   
2019-09-01                0.182206                0.185235        0.172176   
2019-10-01                0.179086                0.182206        0.168668   
2019-11-01                0.176011                0.179086        0.165254   
2019-12-01                0.173140                0.176011        0.161942   

            CPI Core_lag_2  CPI Non-Core_lag_1  CPI Non-Core_lag_2  \
Fecha                                                                
2019-08-01        0.179341            0.123430            0.126055   
2019-09-01        0.175751            0.120649            0.123430   
2019-10-01        0.172176            0.117677            0.120649   
2019-11-01        0.168668            0.114708            0.117677   
2019-12-01        0.165254            0.112035            0.114708   

            CPI Food and Energy_lag_1  ...  Real Minimum Wage (Index)_lag_1  \
Fecha                                  ...                                    
2019-08-01                   0.123753  ...                        -0.002029   
2019-09-01                   0.117414  ...                        -0.000609   
2019-10-01                   0.111047  ...                        -0.000064   
2019-11-01                   0.104757  ...                        -0.001106   
2019-12-01                   0.098706  ...                        -0.001088   

            Real Minimum Wage (Index)_lag_2  Wheat (US$ per ton)_lag_1  \
Fecha                                                                    
2019-08-01                         0.000863                  -0.078909   
2019-09-01                        -0.002029                  -0.071854   
2019-10-01                        -0.000609                  -0.028875   
2019-11-01                        -0.000064                   0.040636   
2019-12-01                        -0.001106                   0.072377   

            Wheat (US$ per ton)_lag_2  Corn  (US$ per ton)_lag_1  \
Fecha                                                              
2019-08-01                   0.071044                   0.011487   
2019-09-01                  -0.078909                  -0.127216   
2019-10-01                  -0.071854                  -0.044102   
2019-11-01                  -0.028875                   0.066893   
2019-12-01                   0.040636                  -0.035344   

            Corn  (US$ per ton)_lag_2  Soybean oil (US$ per ton)_lag_1  \
Fecha                                                                    
2019-08-01                   0.168096                         0.006573   
2019-09-01                   0.011487                         0.016693   
2019-10-01                  -0.127216                         0.019052   
2019-11-01                  -0.044102                         0.042443   
2019-12-01                   0.066893                         0.026200   

            Soybean oil (US$ per ton)_lag_2  Crude oil (US$ per barrel)_lag_1  \
Fecha                                                                           
2019-08-01                         0.042708                          0.048935   
2019-09-01                         0.006573                         -0.044064   
2019-10-01                         0.016693                          0.037436   
2019-11-01                         0.019052             

In [20]:
df.tail()

CPI  CPI Tradable  CPI Non-Tradable  CPI Core  CPI Non-Core  \
Fecha                                                                          
2019-08-01  0.155238      0.107169          0.182206  0.172176      0.120649   
2019-09-01  0.151966      0.103574          0.179086  0.168668      0.117677   
2019-10-01  0.148757      0.100074          0.176011  0.165254      0.114708   
2019-11-01  0.145714      0.096671          0.173140  0.161942      0.112035   
2019-12-01  0.142886      0.093373          0.170545  0.158729      0.109809   

            CPI Food and Energy  CPI Excluding Food and Energy  \
Fecha                                                            
2019-08-01             0.117414                       0.187309   
2019-09-01             0.111047                       0.186615   
2019-10-01             0.104757                       0.185969   
2019-11-01             0.098706                       0.185426   
2019-12-01             0.092980                       0.185001   

            CPI Food and Beverages  CPI Excluding Food and Beverages  \
Fecha                                                                  
2019-08-01                0.076524                          0.207174   
2019-09-01                0.065826                          0.208745   
2019-10-01                0.055385                          0.210242   
2019-11-01                0.045367                          0.211728   
2019-12-01                0.035862                          0.213221   

            CPI Core Excluding Food and Beverages  ...  Wholesale Price Index  \
Fecha                                              ...                          
2019-08-01                               0.194464  ...              -0.013811   
2019-09-01                               0.190797  ...              -0.029031   
2019-10-01                               0.187204  ...              -0.044027   
2019-11-01                               0.183707  ...              -0.058820   
2019-12-01                               0.180317  ...              -0.073404   

            Reserve Requirement Rate  Monetary Policy Rate  \
Fecha                                                        
2019-08-01                  6.446612                  2.50   
2019-09-01                  5.935527                  2.50   
2019-10-01                  6.617785                  2.50   
2019-11-01                  6.069958                  2.25   
2019-12-01                  6.699655                  2.25   

            Circulating Currency Seasonally Adjusted (mill S/)  \
Fecha                                                            
2019-08-01                                           0.007826    
2019-09-01                                           0.008478    
2019-10-01                                           0.003873    
2019-11-01                                           0.006119    
2019-12-01                                           0.029592    

            Net International Reserves (mill $)  Real Minimum Wage (Index)  \
Fecha                                                                        
2019-08-01                             0.022711                  -0.000609   
2019-09-01                            -0.005991                  -0.000064   
2019-10-01                             0.001857                  -0.001106   
2019-11-01                            -0.017871                  -0.001088   
2019-12-01                             0.023139                  -0.002141   

            Wheat (US$ per ton)  Corn  (US$ per ton)  \
Fecha                                                  
2019-08-01            -0.071854            -0.127216   
2019-09-01            -0.028875            -0.044102   
2019-10-01             0.040636             0.066893   
2019-11-01             0.072377            -0.035344   
2019-12-01             0.032523             0.007789   

            Soybean oil (US$ per ton)  Crude oil (US$ per barrel)  
Fecha         